test-data/unit/lib-stub/typing.pyi

```diff
@@ -16,6 +16,7 @@ NamedTuple = 0
 Type = 0
 no_type_check = 0
 ClassVar = 0
+Final = 0
```

Comment:
Yes, we will be able to move to `typing` only with a PEP (can be a short one and/or combined with literal types). I think we can add it to `typing_extensions` even before PEP acceptance. I would propose to follow a strategy similar to what was done for protocols:
* Allow using `Final` in stubs immediately
* After one mypy release cycle fix bugs and design errors and add `Final` to `typing_extensions`. It would be good to have a draft PEP ready at this point.
* After PEP is accepted, move `Final` from `typing_extensions` to `typing`.

If everyone agrees with such schedule, then I can actually also allow using `typing_extensions.Final` (in stubs, because there is no runtime counterpart yet) right in this PR, please let me know. (Otherwise I can do this in a follow-up PR.)

Reply:
> After one mypy release cycle fix bugs and design errors and add Final to typing_extensions. It would be good to have a draft PEP ready at this point.

We can't really advertise a feature in mypy documentation if it doesn't work yet (say, if `Final` and `final` are missing from `typing_extensions`). So looks like we need to do one of these:

1. Update `typing_extensions` before the first mypy release that includes this PR. We should probably mention which version of `typing_extensions` is required to use `Final`/`final` in mypy docs.
2. Add the new definitions to `mypy_extensions` and release the changes before the first mypy release that includes this PR. Update documentation to use `mypy_extensions` in all relevant examples, instead of `typing_extensions`.
3. Postpone merging this PR until we've done one of the above two things.

Also, please add a note here mentioning that this is still an unofficial extension.

I'm not sure if it's important to start using `Final` in stubs immediately. The benefit is pretty minor compared to protocols, as this is primarily about fixing false negatives, and the concept of 'final' doesn't really exist in Python as such, so deciding which things should be final is less obvious. Duck typing, on the other hand, has a long history in Python and is pretty essential for certain APIs.